src/dmd/cpreprocess.d

```diff
@@ -124,5 +124,15 @@ private const(char)[] cppCommand()
             return "sppn.exe";
         }
     }
-    return "cpp";
+    else version (OpenBSD)
+    {
+        // On OpenBSD, we need to use the actual binary /usr/libexec/cpp
+        // rather than the shell script wrapper /usr/bin/cpp ...
+        // Turns out the shell script doesn't really understand -o
+        return "/usr/libexec/cpp";
+    }
+    else
+    {
+        return "cpp";
```

Comment:
this return is no longer part of the `version(Windows)` path and so it has no unconditional return.
```
src\dmd\cpreprocess.d(108): Error: function `dmd.cpreprocess.cppCommand` no `return exp;` or `assert(0);` at end of function
```

Reply:
Ideally not. Cross-compilation.

Reply:
Thanks for the catch. Updated the diff to fix this.